### DISCUSSION TITLE
##[error]TypeError: Cannot read properties of undefined (reading 'includes')

### DISCUSSION THREAD

**Discussion by Suallent** [rfennell/AzurePipelines#1509] [General]
Hi! Thanks so much for providing a task like that for AzureDevOps.
Unfortunately right now creating a single md hasn't worked out so far.

This is what kind of application we have. Not sure if you need this information:
- Our builds in Azure, unless release, are not published anywhere because most of our project only have dependencies to each other. We are simply using Azure Feeds for our own projects
- We are using a inhouse template to generate semantic versioning of our snapshots (they are always 0.0.0-<BRANCHNAME>-SNAPSHOT)

What I want to try out:
- I just want to create a release-note.md of my current branch, so I can test it before we will use it for our release
- My task was to create a seperated pipeline so we run it when necessary
- I tried different settings inside the task and the error stays the same

Here is my build-releasenote.yml :
`

trigger:
  none

stages:
  - stage: GenerateReleaseNotes
    displayName: 'Generate Release Notes'
    jobs:
      - job: GenerateReleaseNotesJob
        displayName: 'Generate Release Notes Job'
        steps:
          - task: XplatGenerateReleaseNotes@4
            displayName: 'Generate Release Notes'
            inputs:
              outputfile: '$(Build.ArtifactStagingDirectory)/releasenotes.md'
              templateLocation: 'PROJECTNAME-docs'
              templatefile: 'build-handlebars-template.md'
              dumpPayloadToConsole: false
              dumpPayloadToFile: false
              replaceFile: True
              getParentsAndChildren: False
              getAllParents: False
              getIndirectPullRequests: True
              checkstages: false
              failOnStderr: true
`

This is the Log in Azure:

Checking for user defined WIQL query
No WIQL query parameters defined
Leaving WI in default order as returned by API
The default artifact for the release was not linked to an Azure DevOps Git Repo Pull Request
Getting all completed Azure DevOps Git Repo PRs in the Team Project IngOne
Getting associated PRs
Enriching known Pull Requests
Checking the CS associated with the PRs to see if they are inturn associated PRs
Total Builds: [0]
Total Commits: [200]
Total Workitems: [21]
Total Related Workitems (Parent/Children): [0]
Total Release Tests: [0]
Total Tests: [0]
Total Manual Test Runs: [0]
Total Manual Test Configurations: [0]
Total Pull Requests: [0]
Total Indirect Pull Requests: [0]
Total Associated Test WI: [0]
Total Published Artifacts: [0]
Total Consumed Artifacts: [0]
Total WIQL Workitems: [0]
Generating the release notes, the are 1 template(s) to process
Using in-line template
##[error]TypeError: Cannot read properties of undefined (reading 'includes')
##[error]TypeError: Cannot read properties of undefined (reading 'includes')
##[error]TypeError: Cannot read properties of undefined (reading 'includes')
Finishing: Generate Release Notes

**Comment by rfennell** [comment]
Your logs shows the task is trying to use an inline template (passed in as a string parameter), this is because you have a problem with your parameters. The **templateLocation** should be set to either **File** or **Inline**, not a folder name, that is include in the **templateFile** if needed

```
task: XplatGenerateReleaseNotes@4
displayName: 'Generate Release Notes'
inputs:
  outputfile: '$(Build.ArtifactStagingDirectory)/releasenotes.md'
  templateLocation: File
  templatefile: 'PROJECTNAME-docs/build-handlebars-template.md'
 ....
``

**Comment by progrock100** [comment]
I just started with this fabulous task and got the same error, so thanks for giving the solution without me having to ask the question !
